src/components/Vehicle.cxx

```diff
@@ -0,0 +1,186 @@
+        
+        bool isFrontWheel = connectionPoints_[id].z_ > 0.0f;
+        wheelNode->SetRotation(connectionPoint.x_ >= 0.0 ? Urho3D::Quaternion(0.0f, 0.0f, -90.0f) : Urho3D::Quaternion(0.0f, 0.0f, 90.0f));
+        wheelNode->SetWorldPosition(node_->GetWorldPosition() + node_->GetWorldRotation() * connectionPoints_[id]);
+        vehicle->AddWheel(wheelNode, wheelDirection, wheelAxle, suspensionRestLength_, wheelRadius_, isFrontWheel);
+        vehicle->SetWheelSuspensionStiffness(id, suspensionStiffness_);
+        vehicle->SetWheelDampingRelaxation(id, suspensionDamping_);
+        vehicle->SetWheelDampingCompression(id, suspensionCompression_);
+        vehicle->SetWheelFrictionSlip(id, wheelFriction_);
+        vehicle->SetWheelRollInfluence(id, rollInfluence_);
+        wheelNode->SetScale(Urho3D::Vector3(0.067f, 0.023f, 0.067f));// tire Diameter: 67mm
+        auto* pWheel = wheelNode->CreateComponent<Urho3D::StaticModel>();
+        pWheel->SetModel(cache->GetResource<Urho3D::Model>("Models/SmartTire.mdl"));
+        pWheel->SetMaterial(cache->GetResource<Urho3D::Material>("Materials/Stone.xml"));
+        pWheel->SetCastShadows(true);
+    }
+    vehicle->ResetWheels();
+}
+void Vehicle::resetDifferential() {
+    auto* vehicle = node_->GetComponent<Urho3D::RaycastVehicle>();
+    engineForce_ = 0;
+    vehicle->SetEngineForce(0, 0);
+    vehicle->SetEngineForce(2, 0);
+    vehicle->SetEngineForce(1, 0);
+    vehicle->SetEngineForce(3, 0);
+}
+void Vehicle::setDifferential(float engineForce, const unsigned int control) {
+    // the Control checks whether you want to turn left or right.
+    // will remove this once i got the Differential.cxx to work
+    auto* vehicle = node_->GetComponent<Urho3D::RaycastVehicle>();
+    if (control == 8) {
+        engineForce = engineForce * -2;
+    } else {
+        engineForce = engineForce * 2;
+    }
+    vehicle->SetEngineForce(1, engineForce);
+    vehicle->SetEngineForce(3, engineForce);
+    vehicle->SetEngineForce(0, -engineForce);
+    vehicle->SetEngineForce(2, -engineForce);
+}
+void Vehicle::FixedUpdate(float timeStep) {
+    float accelerator = 0.0f;
+    auto* vehicle = node_->GetComponent<Urho3D::RaycastVehicle>();
+    // Read controls
+
+    if (controls_.buttons_ & CTRL_LEFT) {
+        setDifferential(maxEngineForce_, CTRL_LEFT);
+    }
+    if (controls_.buttons_ & CTRL_RIGHT) {
+        setDifferential(maxEngineForce_, CTRL_RIGHT);
+    }
+    if (controls_.buttons_ & CTRL_FORWARD) {
+        resetDifferential();
+        accelerator = 0.3f;     // not accurate
+    }
+    if (controls_.buttons_ & CTRL_BACK) {
+        resetDifferential();
+        accelerator = -0.3f;    // not accurate
+    }
+
+    // apply forces
+    engineForce_ = maxEngineForce_ * accelerator;
+   /* vehicle->SetEngineForce(0, engineForce_);
+    vehicle->SetEngineForce(1, engineForce_);*/
+    vehicle->SetEngineForce(3, engineForce_);
+    vehicle->SetEngineForce(2, engineForce_);
+}
```

Comment:
^controls

Reply:
I removed this part, but where should I set the `setEngineForce`?

Reply:
You're already using the trick on how to access the RayCastVehicle from the outside world:
```cpp
auto* vehicle = node_->GetComponent<Urho3D::RaycastVehicle>();
```
If you have access to the node (`m_vehicle_node`) that the `Vehicle` is attached to you can call this method to get  it.

Reply:
yes, but where? I mean where the `setEngineForce()` should be called?

Reply:
It doesnt need to be called, we already call it in WheelServo but if you want to create alternative controls for keyboard that can be done quite easily with this.